src/dra/5D5BC.c

```diff
@@ -754,14 +762,13 @@ s32 HandleDamage(DamageParam* damage, s32 arg1, s32 amount, s32 arg3) {
 // !FAKE: explicitly casting two pointers to s32
 // before comparing them, that's weird
 void func_800FEE6C(void) {
-    s32* var_v1 = D_80139828;
-
+    s32* p = D_80139828;
     do {
-        if (*var_v1 != 0) {
-            (*var_v1)--;
+        if (*p) {
+            (*p)--;
         }
-        var_v1++;
-    } while ((s32)var_v1 < (s32)&D_80139828[0x10]);
+        p++;
+    } while ((long long)p < (long long)&D_80139828[LEN(D_80139828)]);
```

Comment:
This shouldn't block merging, but the convention I've typically seen for comparing to the end of a string is `(str + len(str))` as opposed taking a reference to one past the end of the array. The advantage to the addition is static analyzers will see out of bounds array access and flag it as a bug, when really the goal is to make sure the pointer doesn't go out of bounds to begin with.